재호/17299.py

```diff
@@ -0,0 +1,28 @@
+# https://www.acmicpc.net/problem/17299
+
+from sys import stdin
+
+
+def solution():
+    N = int(stdin.readline())
+    A = list(map(int, stdin.readline().split()))
+    F = [0] * ((10 ** 6) + 1)
+    answer = [0] * (10 ** 6)
+    stk = []
+
+    for x in A:
+        F[x] += 1
+
+    stk.append(0)
+    for i in range(1, len(A)):
+        while stk and F[A[stk[-1]]] < F[A[i]]:
+            answer[stk.pop()] = A[i]
+        stk.append(i)
+
+    while stk:
+        answer[stk.pop()] = -1
+
+    print(" ".join([str(answer[i]) for i in range(N)]))
```

Comment:
```suggestion
    print(*answer)
```
[서영님 코드](https://github.com/Yapp-17th-Algorithm/algorithm-python/pull/9/files#diff-de974c52d9bfba9dfb092bbee362b8afR22)를 보면 이렇게 줄일 수 있을 것 같습니다.

해당 문법은 찾아보니 [언패킹](https://dojang.io/mod/page/view.php?id=2345) 이라고 하네요!

Reply:
엄청 간단하군요.. 좋습니다!